docs: add get started page (#5519)
gnolang/gno: docs/getting-started.md

# Getting started with Gno.land

Gno.land is a Layer 1 blockchain where smart contracts are written in **Gno**,
a deterministic variant of Go. If you can write Go, you can write Gno.

This page takes you from nothing installed to a working toolchain, a key,
test tokens, and your first query against a live network. Plan around
15 minutes.

:::tip Just want to try Gno?
You don't have to install anything to play with Gno code. Open the

> **[Gno Playground](https://play.gno.land)**, write Gno in your browser,

Reviewer: When I click on the Playground link, it is pre-loaded with an example. When I click Run, it doesn't work. It gives the error:
```
panic running expression main(): hello:0:0: name main not declared:
```
What are the steps to make fix this?

Reviewer 2: The run command is looking for the `main()` function by default, but you could change it, there's a popup that allows you to write something else by clicking in the "down arrow" next to run, there you could for example type:
```
println(Render(""))
```

In any case, that's something to improve.

Reviewer: @jeronimoalbi , this may be the first time that a new user interacts with Gno. There are two ways to make the default code example work when they click Run:

* Make the string you suggest be the default value of the Run button: `println(Render(""))`
* Leave the default value of the Run button as `main()` and update the default code example with:
```
func main() {
  println(Render(""))
}
```

What do you think?

Reviewer 2: The second choice makes more sense within playground, but none is ideal 🤔 

Somehow related, check #5549, which aims to be used instead of the current Playground, for better integration.

and run it instantly. Come back here when you're ready to build locally.

> :::
> 
> ## TL;DR
> 
> ```sh
> git clone https://github.com/gnolang/gno.git && cd gno && make install
> gnokey add dev
> # → open https://faucet.gno.land and paste your g1… address
> gnokey query bank/balances/<your-g1-address> -remote https://rpc.gno.land:443
> ```

Reviewer 2: Nit

````suggestion
```sh
git clone https://github.com/gnolang/gno.git && cd gno && make install
gnokey add dev
```

Open [Faucet Hub](https://faucet.gno.land) and paste your "g1…" address to get GNOTs.

```sh
gnokey query bank/balances/<your-g1-address> -remote https://rpc.gno.land:443
```
````

> Then read [Writing Gno code](./builders/anatomy-of-a-gno-package.md).

Reviewer 2: Nit

```suggestion
Then read [Anatomy of a Gno package](./builders/anatomy-of-a-gno-package.md) to learn how to write smart contracts for Gno.land.
```

| Binary   | Purpose                                                  |
|----------|----------------------------------------------------------|
| `gno`    | Gno language CLI — build, test, format, lint `.gno` code |
| `gnokey` | Wallet and transaction CLI — keys, queries, deploys      |
| `gnodev` | Local development node with hot reload                   |

:::info
Make sure `$(go env GOPATH)/bin` is on your `PATH`. If `gno` or `gnokey`
is reported as "command not found", this is almost always the reason.
:::

### Verify the installation

```sh
gno version
gnokey version
gnodev --help    # gnodev has no `version` subcommand; --help confirms it runs
```

## Alternative: Docker

If you prefer not to build from source, official images are published to
GitHub Container Registry:

```sh
docker pull ghcr.io/gnolang/gno            # the gno CLI
docker pull ghcr.io/gnolang/gno/gnokey     # wallet / tx CLI
docker pull ghcr.io/gnolang/gno/gnodev     # local dev node
docker pull ghcr.io/gnolang/gno/gnoland    # full node
docker pull ghcr.io/gnolang/gno/gnoweb     # web frontend
```

Quick smoke test:

```sh
docker run --rm ghcr.io/gnolang/gno version
```

## Create your first key

> `gnokey` stores keys locally. Create a development key named `dev`:

Reviewer 2: ```suggestion
Keys are stored locally by `gnokey`. To create a development key named "dev" run:
```

```sh
gnokey add dev
```

You will be asked for a password, then shown a 24-word mnemonic.

:::warning
Treat any key you create this way as **development-only**. Do not reuse
the mnemonic for real funds. Tokens distributed by the faucet are
testnet tokens.
:::

List your keys to see the address that was generated:

```sh
gnokey list
```

> The address starts with `g1…` — this is how you'll be identified on-chain.

Reviewer 2: Maybe, to distinguish an address from a command notation, if it makes sense:

```suggestion
The address starts with "g1…" — this is how you'll be identified on-chain.
```

If so there other similar address notations to be updated in this document.

## Get test tokens

Test transactions require test tokens (GNOT). The community faucet
distributes them:

→ **[faucet.gno.land](https://faucet.gno.land)**

Paste the `g1…` address from `gnokey list`, select a network, and submit.
Tokens arrive in a few seconds.

## Query a live network

Before deploying anything, confirm your setup can reach a real network.
The following command queries your account balance via ABCI — no
transaction, no gas:

```sh
gnokey query bank/balances/<your-g1-address> \
  -remote https://rpc.gno.land:443
```

If the faucet step worked, the response shows your balance as a
`<amount>ugnot` data field at the current block height.

The main public networks are:

| Network  | RPC endpoint                               | Chain ID   |
|----------|--------------------------------------------|------------|
| Betanet  | `https://rpc.gno.land:443`                 | `gnoland1` |
| Staging  | `https://rpc.staging.gno.land:443`         | `staging`  |
| Test11   | `https://rpc.test11.testnets.gno.land:443` | `test11`   |

> See [Networks](./resources/gnoland-networks.md) for the full reference.

Reviewer 2: This would avoid duplication between docs:

```suggestion
See the list of [Networks](./resources/gnoland-networks.md) for the full reference.
```

## Before you deploy

Two things to know before publishing code on-chain.

### Namespaces

Today, anyone can deploy under their own address-based namespace without
registration. Username-based namespaces like `gno.land/r/alice/...` are **not available
yet** and will require a registration step.

### License agreement

Deploying code on gno.land is a public act: your code runs on a shared
network and others may build on top of it. To make the terms clear,
gno.land may require you to accept a **Contributor License Agreement (CLA)**
before you can publish packages.

If your first `gnokey maketx addpkg` is rejected with
`has not signed the required CLA`, sign once by calling the on-chain
`r/sys/cla` realm with the current hash:

```sh
gnokey maketx call \
  -pkgpath gno.land/r/sys/cla -func Sign -args "<current-hash>" \
  -gas-fee 100000ugnot -gas-wanted 2000000 \
  -broadcast -chainid <chain-id> -remote <rpc-endpoint> \
  dev
```

> For Betanet: `-chainid gnoland1 -remote https://rpc.gno.land:443`.

Reviewer 2: ```suggestion
:::info
For Betanet use `-chainid gnoland1 -remote https://rpc.gno.land:443`.
:::
```

Get the current `<hash>` at [`gno.land/r/sys/cla`](https://gno.land/r/sys/cla).

> Canonical text: [`CLA.md`](https://github.com/gnolang/gno/blob/master/CLA.md).

Reviewer 2: ```suggestion
Before signing read the [Contributor License Agreement](https://github.com/gnolang/gno/blob/master/CLA.md)
and then get the current `<hash>` at [`gno.land/r/sys/cla`](https://gno.land/r/sys/cla).
```

# Getting started with Gno.land

> Gno.land is a Layer 1 blockchain where smart contracts are written in **Gno**,

Reviewer 3: You should minima reference docs/builders/what-is-gnolang.md

a deterministic variant of Go. If you can write Go, you can write Gno.

This page takes you from nothing installed to a working local chain
with `gnodev`, then shows how to graduate to a live network when you're

> ready to deploy. Plan around 15 minutes.

Reviewer 3: This is a good idea, but this can be added as a generic subtitle for each documentation

:::tip Just want to try Gno?

> You don't have to install anything to play with Gno code. Open the

Reviewer 3: > Try the **[Playground](https://play.gno.land)** to write Gno in your browser

**[Gno Playground](https://play.gno.land)**, write Gno in your browser,
and run it instantly. Come back here when you're ready to build locally.
:::

## TL;DR

> ```sh

Reviewer 3: I think for this block code you should:
- Have a comment for each command executed describing what it precisely does (# Install gnoland # Run local node and Gnoweb instance)
- Avoid semantically non-essential words (e.g.: `Later, to go live` -> ``Add a key name `dev` ``)
- Keep one precise goal for each code block (Run local environment, interact with networks, ...)
- Add a title for each block 
- Avoid using betanet / mainnet for testing purpose (gnodev, then `-remote https://rpc.gno.land:443` some line later)

The end goal is to have a self dependent TL;DR usable by anyone (beginner, advanced user that want to remind a specific command, ...).

> git clone https://github.com/gnolang/gno.git && cd gno && make install

Reviewer 3: I know this is in the pipe, but this is dependant of: https://github.com/gnolang/gno/pull/5492

> git clone https://github.com/gnolang/gno.git && cd gno && make install

Reviewer 3: You shouldn't add the installation in TL;DR. It is dependent on user situation, it is better to redirect to installation section or have a one line installer directly referenced.
See: https://github.com/gnolang/gno/pull/5519/changes#r3099343803
Or see: https://go.dev/dl/

gnodev
# → open http://localhost:8888 and start building

# Later, to go live:
gnokey add dev
# → open https://faucet.gno.land and paste your g1… address
gnokey query bank/balances/<your-g1-address> -remote https://rpc.gno.land:443
```

> Then read [Anatomy of a Gno package](./builders/anatomy-of-a-gno-package.md) to learn how to write smart contracts for Gno.land.

Reviewer 3: ```suggestion
- [Anatomy of a Gno package](./builders/anatomy-of-a-gno-package.md): Learn how to write Gno packages (`/p`, `/r`)
```

## Prerequisites

- **Git**

> - **[Go 1.24+](https://go.dev/dl/)** — required by the repository's `go.mod`

Reviewer 3: I think we generally want to keep latest version of Go. It would be more efficient to have a generic statement with a link to `go.mod` if that's the case.
Otherwise we need to keep in mind to update that regularly.

- **Make**
- Linux and macOS are the primary supported platforms. Windows users
  should develop inside [WSL2](https://learn.microsoft.com/windows/wsl/install).

## Install from source

```sh

> git clone https://github.com/gnolang/gno.git

Reviewer 3: You should have a section `Installation` with collapsible section {`One-line installer`, `Install from source`, `Binary releases`, `Docker`}. 
I don't like reading a tutorial, and getting twice the same information in the first 1min of reading (TL;DR + first paragraph).

IMO installation should even have its own page, as download can be very dependent on user machine.
If you want to make a very detailed installation, you should make a `step by step` (directed for beginner), and other docs reference the installation docs

cd gno
make install
```

> The first build takes a few minutes while Go downloads modules. When it

Reviewer 3: I think this is not important to know

finishes, you should see three confirmation lines:

```
[+] 'gnokey' has been installed. Read more in ./gno.land/
[+] 'gno' has been installed. Read more in ./gnovm/
[+] 'gnodev' has been installed. Read more in ./contribs/gnodev/
```

This builds and installs three binaries into `$(go env GOPATH)/bin`:

| Binary   | Purpose                                                  |
|----------|----------------------------------------------------------|
| `gno`    | Gno language CLI — build, test, format, lint `.gno` code |
| `gnokey` | Wallet and transaction CLI — keys, queries, deploys      |
| `gnodev` | Local development node with hot reload                   |

:::info
Make sure `$(go env GOPATH)/bin` is on your `PATH`. If `gno` or `gnokey`
is reported as "command not found", this is almost always the reason.
:::

### Verify the installation

```sh
gno version
gnokey version

> gnodev --help    # gnodev has no `version` subcommand; --help confirms it runs

Reviewer 3: ```suggestion
gnodev --help
```

Reviewer 3: And this should be fixed, why gnodev don't have version subcommand?

transaction, no gas:

```sh
gnokey query bank/balances/<your-g1-address> \
  -remote https://rpc.gno.land:443
```

If the faucet step worked, the response shows your balance as a
`<amount>ugnot` data field at the current block height.

See the list of [Networks](./resources/gnoland-networks.md) for the full reference.

## Before you deploy

Two things to know before publishing code on-chain.

### Namespaces

Today, anyone can deploy under their own address-based namespace without
registration. Username-based namespaces like `gno.land/r/alice/...` are **not available
yet** and will require a registration step.

### License agreement

Deploying code on gno.land is a public act: your code runs on a shared
network and others may build on top of it. To make the terms clear,
gno.land may require you to accept a **Contributor License Agreement (CLA)**
before you can publish packages.

If your first `gnokey maketx addpkg` is rejected with
`has not signed the required CLA`, sign once by calling the on-chain
`r/sys/cla` realm with the current hash:

```sh
gnokey maketx call \
  -pkgpath gno.land/r/sys/cla -func Sign -args "<current-hash>" \
  -gas-fee 100000ugnot -gas-wanted 2000000 \

>   -broadcast -chainid <chain-id> -remote <rpc-endpoint> \

Reviewer: ```suggestion
  -chainid <chain-id> -remote <rpc-endpoint> \
```
We just merged the PR for gnokey to broadcast by default.

Author: @jefft0 awesome your PR was part of our UX task list !
